### PR TITLE
React & Typescript Support

### PR DESCRIPTION
**Pull request by gregdaynes**
Provides support for using React with Typescript to create an application
- Defines minimal typescript config to transpile down to es5
- Configures Webpack to handle `.js` `.jsx` `.ts` `.tsx` files
- Allow for app.css to be ignored by typescript for CSS Preprocessing
- Adds small demo app for React

### CODE REVIEW

File: assets/js/app.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
-// We need to import the CSS so that webpack will load it.
+// We need to require the CSS so that Typescript will igore
```
Comment (hez) [inline assets/js/app.js]: _ignore_

Comment (gregdaynes) [reply assets/js/app.js]: Igor!

**Comment by gregdaynes** [comment]
Fixed typo in CSS require comment.

**Comment by gregdaynes** [comment]
Rebased with Master to resolve merge conflict in yarn.lock

**Pull request merged**